floodviz/map_utils.py

```diff
@@ -2,15 +2,33 @@
 
 def site_dict(site_list, url_prefix):
     """Puts site data into a dictionary
-
     Args:
         site_list: A list of site ids to be queried on NWIS
-        url_prefix: The beginning of the NWIS url
-
+        url_prefix: A string containing the beginning of the NWIS site url
     Returns:
-        A dict containing various site information in a usable format.
+        An array of dicts containing various site information in a usable format.
     """
 
+    if not site_list:
+        print("Error: site_list is empty")
+        return
+
+    if not url_prefix:
+        print("Error: url_prefix is empty")
+        return
+
+    for site in site_list:
+        if not site.isdigit():
+            print("Error: list contains a non-numeric element")
+            return
+        if len(site) != 8:
+            print("Error: all site ids must be 8 digits")
+            return
+
+    if not url_prefix == 'https://waterservices.usgs.gov/nwis/site':
```

Comment:
This is unnecessary. The point of passing in the url_prefix is so that the endpoint can be changed.